clippy_config/src/types.rs

```diff
@@ -23,12 +29,34 @@ impl DisallowedPath {
         path
     }
 
+    pub fn diag_amendment(&self, span: Span) -> impl FnOnce(&mut Diag<'_, ()>) + use<'_> {
+        move |diag| {
+            if let Some(replacement) = self.replacement() {
+                diag.span_suggestion(
+                    span,
+                    self.reason().map_or_else(|| String::from("use"), ToOwned::to_owned),
+                    replacement,
+                    Applicability::MachineApplicable,
+                );
+            } else if let Some(reason) = self.reason() {
```

Comment:
Why don't we always show the reason?

Reply:
My thinking was: if a reason is provided, show that; otherwise, show:
```
use: `<replacement>`
```

Reply:
Ohh, yeah missed that it's shown in the suggestion anyway. Sounds good then :+1: